sno/diff_output.py

```diff
@@ -392,19 +408,23 @@ def _out(dataset, diff):
 class LazyGeojsonFeatureOutput:
     """Wrapper of KeyValue that lazily serialises it as GEOJSON when sent to json.dumps"""
 
-    __slots__ = ("change_type", "key_value")
+    __slots__ = ("change_type", "key_value", "geometry_transform")
```

Comment:
isn't there one transform per dataset? Do we need to carry it around with every feature?

Reply:
yes, there's only one of these objects (all features get the same one).

but the geometry isn't turned into an OGR geometry until it's being streamed out as JSON. So we need to pass the transform down into the lazily-serialized object so it can be used after we have an OGR geometry.

Doing it this way is quite elegant because it ensures minimal memory overhead; we don't have to store millions of OGR geometries in memory (we just create one at a time while streaming JSON out)

Reply:
sure, I doing the transform at emit-time makes total sense. But why do we need to pass N million pointers to the same geometry_transform object about with each feature?

Reply:
the alternative being a global variable? I try to avoid those where it's not absolutely necessary

Reply:
Create an output context containing that?

I'm not familiar enough with how this is working though, I'd have to dig in tomorrow — eg: I thought we were generating/producing a feature at a time, but that doesn't seem to be the case, we're building up big features lists.

Reply:
sort of - we're building up lists of dicts, but only transforming & converting during `json.dump()`. It's a lot less work up-front than it used to be.